test/jobworker/jobworker_test.go

```diff
@@ -0,0 +1,299 @@
+package jobworker
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/tjper/teleport/internal/encrypt"
+	pb "github.com/tjper/teleport/proto/gen/go/jobworker/v1"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+var port = flag.Int("port", 8080, "port jobworker API is serving content on")
```

Comment:
Flags in tests are a pain, in my experience. They force you to single out tests in CI so you can set the flags, which causes all sorts of complications. Better to use env variables.

In this case we should be using the equivalent of net.Listen("localhost:") to bring up a new server, instead of picking an explicit port number.